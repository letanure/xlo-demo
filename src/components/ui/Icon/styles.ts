import styled, { css } from 'styled-components'

import { IconProps } from '.'

export const Wrapper = styled.span<IconProps>`
  ${({ name, width }) => css`
    background-image: url(/img/icons/${name}.svg);
    width: ${width}px;
    height: ${width}px;
    display: inline-block;
    background-size: cover;
    background-position: center;
    background-repeat: no-repeat;
  `}
`
