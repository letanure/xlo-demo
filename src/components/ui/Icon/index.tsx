import * as S from './styles'

type IconNames = 'info'

export type IconProps = {
  name: IconNames
  width?: number
}

const Icon = ({ name, width = 20 }: IconProps) => (
  <S.Wrapper width={width} name={name}></S.Wrapper>
)

export default Icon
