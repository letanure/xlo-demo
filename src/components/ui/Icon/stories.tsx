import { ComponentStory, ComponentMeta } from '@storybook/react'

import Icon from '.'

export default {
  title: 'Layout/Icon',
  component: Icon,
  argTypes: {
    name: { control: 'text' },
    width: { control: 'number' }
  }
} as ComponentMeta<typeof Icon>

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} />

export const Basic = Template.bind({})
Basic.args = {
  name: 'info'
}

export const Width = Template.bind({})
Width.args = {
  name: 'info',
  width: 48
}
