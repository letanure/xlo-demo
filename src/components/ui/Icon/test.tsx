import { render, screen } from '@testing-library/react'

import Icon from '.'

describe('<Icon />', () => {
  it.skip('should render something', () => {
    const { container } = render(<Icon />)

    expect(screen.getByRole('heading', { name: /Icon/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
