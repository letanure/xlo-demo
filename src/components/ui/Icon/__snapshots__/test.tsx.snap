// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render something 1`] = `
.c0 {
  content: 'default';
}

<div
  class="c0"
>
  <h1>
    Icon with 
    default
  </h1>
</div>
`
